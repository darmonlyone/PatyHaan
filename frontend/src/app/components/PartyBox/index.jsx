import React from "react";

import StatusButton from "app/components/StatusButton";
import NoImage from "app/asset/noimage.png";

const PartyBox = ({ id, imageUrl, title, joined, amount, status }) => {
	return (
		<div className="flex flex-col justify-center w-44 lg:w-52 m-1 sm:m-2 lg:m-3 shadow-md border border-gray-100 rounded-md bg-white">
			<img
				data-testid="image"
				className="w-auto h-44 lg:h-52 rounded-t-md"
				src={imageUrl || NoImage}
				alt={imageUrl || "No Image"}
			/>
			<span className="p-2 text-md lg:text-lg max2line h-14 lg:h-16">
				{title}
			</span>
			<div className="border-t border-gray-300 flex flex-row p-2">
				<span className="flex w-2/5 text-sm justify-start items-center">{`${joined}/${amount}`}</span>
				<StatusButton id={id} status={status} />
			</div>
		</div>
	);
};

export default PartyBox;
