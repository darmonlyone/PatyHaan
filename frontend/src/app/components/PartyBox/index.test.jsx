import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";

import PartyBox from ".";

it("should render PartyBox with default value", async () => {
	const { asFragment } = render(
		<Router>
			<PartyBox />
		</Router>
	);

	expect(asFragment()).toMatchSnapshot();
});

it("should render PartyBox with some value", async () => {
	const { asFragment, getByTestId,getByText } = render(
		<Router>
			<PartyBox id="2" imageUrl="test url" title="test" joined="2" amount="10"/>
		</Router>
	);

	expect(asFragment()).toMatchSnapshot();
	expect(getByTestId("image").src).toEqual("http://localhost/test%20url")
	expect(getByText("test")).toBeDefined()
	expect(getByText("2/10")).toBeDefined()

});

it("should render PartyBox with status joined", async () => {
	const { asFragment,getByText } = render(
		<Router>
			<PartyBox id="2" imageUrl="test url" title="test" joined="2" amount="10" status="joined"/>
		</Router>
	);
	
	expect(asFragment()).toMatchSnapshot();
	expect(getByText("ออกปาร์ตี้")).toBeDefined()
});

it("should render PartyBox with status full", async () => {
	const { asFragment,getByText } = render(
		<Router>
			<PartyBox id="2" imageUrl="test url" title="test" joined="2" amount="10" status="full"/>
		</Router>
	);
	
	expect(asFragment()).toMatchSnapshot();
	expect(getByText("เต็ม")).toBeDefined()
});

it("should render PartyBox with status owned", async () => {
	const { asFragment,getByText } = render(
		<Router>
			<PartyBox id="2" imageUrl="test url" title="test" joined="2" amount="10" status="owned"/>
		</Router>
	);
	
	expect(asFragment()).toMatchSnapshot();
	expect(getByText("ลบปาร์ตี้")).toBeDefined()
});


it("should render PartyBox with status none", async () => {
	const { asFragment,getByText } = render(
		<Router>
			<PartyBox id="2" imageUrl="test url" title="test" joined="2" amount="10" status="none"/>
		</Router>
	);
	
	expect(asFragment()).toMatchSnapshot();
	expect(getByText("เข้าร่วม")).toBeDefined()
});


