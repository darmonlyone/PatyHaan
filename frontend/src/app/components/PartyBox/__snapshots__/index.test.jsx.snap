// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PartyBox with default value 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col justify-center w-44 lg:w-52 m-1 sm:m-2 lg:m-3 shadow-md border border-gray-100 rounded-md bg-white"
  >
    <img
      alt="No Image"
      class="w-auto h-44 lg:h-52 rounded-t-md"
      data-testid="image"
      src="noimage.png"
    />
    <span
      class="p-2 text-md lg:text-lg max2line h-14 lg:h-16"
    />
    <div
      class="border-t border-gray-300 flex flex-row p-2"
    >
      <span
        class="flex w-2/5 text-sm justify-start items-center"
      >
        undefined/undefined
      </span>
      <button
        class="w-3/5 text-sm py-1"
      >
        เข้าร่วม
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`should render PartyBox with some value 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col justify-center w-44 lg:w-52 m-1 sm:m-2 lg:m-3 shadow-md border border-gray-100 rounded-md bg-white"
  >
    <img
      alt="test url"
      class="w-auto h-44 lg:h-52 rounded-t-md"
      data-testid="image"
      src="test url"
    />
    <span
      class="p-2 text-md lg:text-lg max2line h-14 lg:h-16"
    >
      test
    </span>
    <div
      class="border-t border-gray-300 flex flex-row p-2"
    >
      <span
        class="flex w-2/5 text-sm justify-start items-center"
      >
        2/10
      </span>
      <button
        class="w-3/5 text-sm py-1"
      >
        เข้าร่วม
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`should render PartyBox with status full 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col justify-center w-44 lg:w-52 m-1 sm:m-2 lg:m-3 shadow-md border border-gray-100 rounded-md bg-white"
  >
    <img
      alt="test url"
      class="w-auto h-44 lg:h-52 rounded-t-md"
      data-testid="image"
      src="test url"
    />
    <span
      class="p-2 text-md lg:text-lg max2line h-14 lg:h-16"
    >
      test
    </span>
    <div
      class="border-t border-gray-300 flex flex-row p-2"
    >
      <span
        class="flex w-2/5 text-sm justify-start items-center"
      >
        2/10
      </span>
      <button
        class="w-3/5 text-sm py-1 bg-gray-400"
        disabled=""
      >
        เต็ม
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`should render PartyBox with status joined 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col justify-center w-44 lg:w-52 m-1 sm:m-2 lg:m-3 shadow-md border border-gray-100 rounded-md bg-white"
  >
    <img
      alt="test url"
      class="w-auto h-44 lg:h-52 rounded-t-md"
      data-testid="image"
      src="test url"
    />
    <span
      class="p-2 text-md lg:text-lg max2line h-14 lg:h-16"
    >
      test
    </span>
    <div
      class="border-t border-gray-300 flex flex-row p-2"
    >
      <span
        class="flex w-2/5 text-sm justify-start items-center"
      >
        2/10
      </span>
      <button
        class="w-3/5 text-sm py-1 bg-blue-400"
      >
        ออกปาร์ตี้
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`should render PartyBox with status none 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col justify-center w-44 lg:w-52 m-1 sm:m-2 lg:m-3 shadow-md border border-gray-100 rounded-md bg-white"
  >
    <img
      alt="test url"
      class="w-auto h-44 lg:h-52 rounded-t-md"
      data-testid="image"
      src="test url"
    />
    <span
      class="p-2 text-md lg:text-lg max2line h-14 lg:h-16"
    >
      test
    </span>
    <div
      class="border-t border-gray-300 flex flex-row p-2"
    >
      <span
        class="flex w-2/5 text-sm justify-start items-center"
      >
        2/10
      </span>
      <button
        class="w-3/5 text-sm py-1"
      >
        เข้าร่วม
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`should render PartyBox with status owned 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col justify-center w-44 lg:w-52 m-1 sm:m-2 lg:m-3 shadow-md border border-gray-100 rounded-md bg-white"
  >
    <img
      alt="test url"
      class="w-auto h-44 lg:h-52 rounded-t-md"
      data-testid="image"
      src="test url"
    />
    <span
      class="p-2 text-md lg:text-lg max2line h-14 lg:h-16"
    >
      test
    </span>
    <div
      class="border-t border-gray-300 flex flex-row p-2"
    >
      <span
        class="flex w-2/5 text-sm justify-start items-center"
      >
        2/10
      </span>
      <button
        class="w-3/5 text-sm py-1 bg-red-400"
      >
        ลบปาร์ตี้
      </button>
    </div>
  </div>
</DocumentFragment>
`;
